### PR TITLE
ability to enter count of number of day to choice releases

### PR DESCRIPTION
**Pull request by monoskribt**
ability to enter count of number of day to choice releases and also log info about returning and throwing state. Added storage playlist of user and content of playlist into database as well.

### CODE REVIEW

File: src/main/java/com/spotifyapi/controller/SpotifyController.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import lombok.AllArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Scheduled;
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/controller/SpotifyController.java:8]: -

File: src/main/java/com/spotifyapi/model/SpotifyTrackFromPlaylist.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.spotifyapi.model;
+
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.UUID;
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/model/SpotifyTrackFromPlaylist.java:7]: -

File: src/main/java/com/spotifyapi/model/SpotifyUserPlaylist.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.spotifyapi.model;
+
+
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.List;
+import java.util.UUID;
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/model/SpotifyUserPlaylist.java:9]: -

File: src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java [hunk 4]
```diff
@@ -80,17 +86,26 @@ public List<PlaylistSimplified> getOfUserPlaylists() {
...
+            var tracks = spotifyApi.getAlbumsTracks(album.getId())
                     .build()
                     .execute()
                     .getItems();
-            Arrays.stream(track).map(TrackSimplified::getUri)
-                    .forEach(trackUrl::add);
+
+            for (TrackSimplified track : tracks) {
+                Track trackFormat = spotifyTrackService.convertToTrackFormat(track);
+                if(!spotifyTrackService.isAlreadyExist(track.getId(), playlist.get().getId())) {
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java:111]: decrease a number of calls to database

File: src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java [hunk 5]
```diff
@@ -119,6 +134,10 @@ public void deleteAllOfTracksFromPlaylistById(String playlistId) {
 
             try {
                 spotifyApi.removeItemsFromPlaylist(playlistId, removeTracks).build().execute();
+
+                List<SpotifyTrackFromPlaylist> listToRemoveTracks = trackRepository
+                        .findAllByUserPlaylistId(playlistId);
+                trackRepository.deleteAll(listToRemoveTracks);
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java:148]: tracksToRemove

File: src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    private final TrackRepository trackRepository;
+    private final PlaylistRepository playlistRepository;
+
+    @Override
+    public void saveTracks(Track track, SpotifyUserPlaylist playlist) {
+        SpotifyTrackFromPlaylist trackFromPlaylist = new SpotifyTrackFromPlaylist();
+        Optional<SpotifyUserPlaylist> spotifyUserPlaylist = playlistRepository.findById(playlist.getId());
+
+
+        if(!isAlreadyExist(track.getId(), spotifyUserPlaylist.get().getId())) {
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java:28]: decrease a number of calls to database

File: src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            trackFromPlaylist.setId(track.getId());
+            trackFromPlaylist.setName(track.getName());
+            trackFromPlaylist.setExternalUrl(track.getExternalUrls().get("spotify"));
+            trackFromPlaylist.setArtistName(track.getArtists()[0].getName());
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java:32]: NPE?

File: src/main/java/com/spotifyapi/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -2,39 +2,106 @@
...
+                        .orElseThrow(() -> new IllegalStateException("Playlist not found after save"));
+
+                var playlistTracks = Arrays.stream(
+                        spotifyApi.getPlaylistsItems(playlist.getId())
+                                .build()
+                                .execute()
+                                .getItems())
+                        .toList();
+
+                for (PlaylistTrack trackItem : playlistTracks) {
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/service/impl/UserServiceImpl.java:71]: decrease a number of calls to database

Comment (syevtushok) [reply src/main/java/com/spotifyapi/service/impl/UserServiceImpl.java:71]: https://www.postgresql.org/docs/current/sql-insert.html OR list

File: src/main/java/com/spotifyapi/controller/SpotifyController.java (hunk 1 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/controller/SpotifyController.java:8]: Done

File: src/main/java/com/spotifyapi/model/SpotifyTrackFromPlaylist.java (hunk 2 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/model/SpotifyTrackFromPlaylist.java:7]: Done

File: src/main/java/com/spotifyapi/model/SpotifyUserPlaylist.java (hunk 3 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/model/SpotifyUserPlaylist.java:9]: Done

File: src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java (hunk 4 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java:111]: Done

File: src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java (hunk 5 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java:148]: Done

File: src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java (hunk 6 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java:28]: Done

File: src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java (hunk 7 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java:32]: Resolved problem with help to check track is null

File: src/main/java/com/spotifyapi/service/impl/UserServiceImpl.java (hunk 8 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/service/impl/UserServiceImpl.java:71]: Resolved

**Pull request merged**